presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -45,23 +47,22 @@ public static Slice decompress(CompressionCodecName codec, Slice input, int unco
             case GZIP:
                 return decompressGzip(input, uncompressedSize);
             case SNAPPY:
-                return decompressSnappy(input, uncompressedSize);
+                return decompress(new SnappyDecompressor(), input, uncompressedSize);
+            case LZO:
+                return decompress(new LzoDecompressor(), input, uncompressedSize);
```

Comment:
Stepping through hadoop lzo impl. (which implements LZO1X) I noticed something different than aircompressor lzo (I don't really know whether aircompressor implements the same algorihtm). Hadoop lzo at the beginning reads two integers from the input stream (4 bytes for original block size + 4 bytes for compressed chunk length) then the rest of the stream is interpreted as compressed data, and it succeeds. I did the same and consumed 8 bytes before passing the data to aircompressor's lzo decompressor, then most of the test cases passed (there were still failures).

Anyway [here](https://gist.github.com/nezihyigitbasi/abec5673fb41e8173d042a73185847cd) is the minimal code that shows how airlift lzo decompressor fails while hadoop's lzo decomp. succeeds with the same input. @dain any comments appreciated.